### PR TITLE
Feature/add reuse key for cluster volumes to cmk

### PR DESCRIPTION
**Pull request by johnc1996**
- Added new field to Terraform resource `mws_customer_managed_key` to stop the key from being used to encrypt cluster volumes
- The new field defaults to `true` to preserve the existing behaviour of this resource so people don't have to update their code to include the new field if they don't want to use it
- Added test to check that setting the field to `false` returns the correct response

### CODE REVIEW

File: mws/resource_customer_managed_key.go [hunk 1]
```diff
@@ -19,11 +19,12 @@ type AwsKeyInfo struct {
...
-	AwsKeyInfo           *AwsKeyInfo `json:"aws_key_info" tf:"force_new"`
-	AccountID            string      `json:"account_id" tf:"force_new"`
-	CreationTime         int64       `json:"creation_time,omitempty" tf:"computed"`
-	UseCases             []string    `json:"use_cases"`
+	CustomerManagedKeyID      string      `json:"customer_managed_key_id,omitempty" tf:"computed"`
+	AwsKeyInfo                *AwsKeyInfo `json:"aws_key_info" tf:"force_new"`
+	AccountID                 string      `json:"account_id" tf:"force_new"`
+	CreationTime              int64       `json:"creation_time,omitempty" tf:"computed"`
+	UseCases                  []string    `json:"use_cases"`
+	ReuseKeyForClusterVolumes bool        `json:"reuse_key_for_cluster_volumes,omitempty" tf:"default:true"`
```
Comment (nfx) [inline mws/resource_customer_managed_key.go:27]: please bump the schema version to 2 and do schema version upgrader:

https://github.com/databrickslabs/terraform-provider-databricks/blob/bb43dd35d51870538e6670982d31f73ffb7d6e17/mws/resource_customer_managed_key.go#L103-L157

Comment (johnc1996) [reply mws/resource_customer_managed_key.go:27]: Version 1 or 2 as current one looks to be 0? Are there docs on what the schema version upgrader is and how I can do that please?

Comment (johnc1996) [reply mws/resource_customer_managed_key.go:27]: Ah was looking at the version within which is at 0. Will update schema version to 2

Comment (johnc1996) [reply mws/resource_customer_managed_key.go:27]: Hi, still wanting to check how I do a schema version upgrader?

Comment (nfx) [reply mws/resource_customer_managed_key.go:27]: @johnc1996 Please see the link in the first comment =)

Comment (johnc1996) [reply mws/resource_customer_managed_key.go:27]: Have pushed a fix now and have passing tests

**Comment by johnc1996** [comment]
When I have tested this in a Terraform project when it is set to `true` it is causing the field to be added to the resource which causes the resource to be rebuilt and in turn it requires the Databricks workspace to be rebuilt. Can you offer any assistance on how to achieve this addition of not having the customer managed key be used to encrypt cluster volumes without introducing breaking changes?

**Comment by nfx** [comment]
@johnc1996 Please add state upgrader, so that workspace is not rebuilt.

**Comment by nfx** [comment]
@johnc1996 Please add state upgrader, so that workspace is not rebuilt.

**Comment by nfx** [comment]
PR is out of date. please create a new one.

**Pull request closed without merging**